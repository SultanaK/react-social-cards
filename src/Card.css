.card-wrapper {
    max-width: 300px;
    display: inline-block;
}

.card {
    background: white;
    color: black;
    height: 550px;        
    margin: .5rem;
    border-radius: 3px;
    overflow: hidden;
}