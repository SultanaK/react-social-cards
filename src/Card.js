import React from 'react'

function Card(props) {
    return (
        <div className="card">
            <img src={props.imageUrl} alt="profile pic" />
            <div className="card-content">
                <h3>Name: {props.name}</h3>
                <p>Age: {props.age}</p>
                <p>Likes: {props.likes}</p>
                <p>{props.description}</p>

            </div>
        </div>
    )
}

export default Card

Card.defaultProps = {
    imageUrl: "https://image.com", 
    name: "Todd",
    age: 5, 
    description: "Lorem ipsum"
}