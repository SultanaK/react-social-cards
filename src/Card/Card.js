import React from 'react'
import { Link } from 'react-router-dom'
import LikeButton from '../LikeButton/LikeButton'
import AppContext from '../AppContext'

class Card extends React.Component {
    static contextType = AppContext;

    render() {
        console.log(this.context)
        return (
            <div className="card">
                <img src={this.props.imageUrl} alt="profile pic" />
                <div className="card-content">
                    <h3>Name: {this.props.name}</h3>
                    <p><Link to={`/profile/${this.props.id}`}>View</Link></p>
                    <LikeButton addLike={this.props.addLike} />
                    <p>Age: {this.props.age}</p>
                    <p>Likes: {this.props.likes}</p>
                    <p>{this.props.description}</p>
                </div>
            </div>
        )
    }
}

export default Card

Card.defaultProps = {
    imageUrl: "https://image.com", 
    name: "Todd",
    age: 5, 
    description: "Lorem ipsum"
}