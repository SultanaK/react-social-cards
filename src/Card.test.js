import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Card from './Card';

const mockData = {
    id: 1,
    name: "Steve",
    imageUrl: "https://source.unsplash.com/mx0DEnfYxic/300x300",
    age: 5,
    description: "Lorem ipsum dolor sit amet"
}

describe('<Card />', () => {

    it('renders without crashing', () => {
        // Create a DOM element to render the component into
        const div = document.createElement('div');

        // Render the component
        // If something is wrong it will fail here
        ReactDOM.render(
        <Card 
            id={mockData.id} 
            imageUrl={mockData.imageUrl}
            description={mockData.desciption}
            name={mockData.name}
            age={mockData.age}
        />, 
        div);

        // Clean up
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders UI as expected when passed props', () => {
        // Render the component, as JSON
        const tree = renderer.create(
        <Card 
            id={mockData.id} 
            imageUrl={mockData.imageUrl}
            description={mockData.desciption}
            name={mockData.name}
            age={mockData.age}
        />
        ).toJSON();
        // Check whether it matches the previous snapshot
        // Stored in __snapshots__/App.test.js.snap
        expect(tree).toMatchSnapshot();  
    });

    it('renders UI as expected without props', () => {
        // Render the component, as JSON
        const tree = renderer.create(
        <Card />
        ).toJSON();
        // Check whether it matches the previous snapshot
        // Stored in __snapshots__/App.test.js.snap
        expect(tree).toMatchSnapshot();  
    });
});