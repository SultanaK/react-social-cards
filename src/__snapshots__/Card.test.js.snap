// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders markup as expected WITH props 1`] = `
<div
  className="card"
>
  <img
    alt="profile pic"
    src="https://image.com"
  />
  <div
    className="card-content"
  >
    <h3>
      Name: 
      Todd
    </h3>
    <p>
      Age: 
      5
    </p>
    <p>
      Lorem ipsum
    </p>
  </div>
</div>
`;

exports[`<Card /> renders markup as expected without props 1`] = `
<div
  className="card"
>
  <img
    alt="profile pic"
    src="https://image.com"
  />
  <div
    className="card-content"
  >
    <h3>
      Name: 
      Todd
    </h3>
    <p>
      Age: 
      5
    </p>
    <p>
      Lorem ipsum
    </p>
  </div>
</div>
`;
